hil-test/tests/init.rs

```diff
@@ -107,4 +107,22 @@ mod tests {
         let delay = Delay::new();
         delay.delay(2000.millis());
     }
+
+    #[test]
+    #[timeout(3)]
+    fn test_init_then_try_init() {
+        esp_hal::init(Config::default());
+
+        esp_hal::try_init(Config::default());
+
+        let delay = Delay::new();
+        delay.delay(2000.millis());
```

Comment:
This just wastes time, you don't need a random delay to show that try_init doesn't panic.